### PR TITLE
Revamp Hero Banner template to new standards and looks

### PR DESCRIPTION
**Pull request by nriserEG**
## Global Template: Hero Banner with Call-To-Action

JIRA: https://help.evergage.com/browse/EVG-11429
This global template replaces a hero banner with a new banner containing a background image and a call-to-action button.

@avkieu @dcohenEG 
I decided to use semantic elements like h1 instead of div (as I originally had planned to do, using ARIA attributes to make them accessible). I’ve styled the template such that it works across various client sites. I tested the template HTML and CSS on sites such as Carhartt, Cisco, Neiman Marcus, Zumiez, Salesforce, and Cumulus and the template renders consistently (aside from font family and rem-based font size).

As for Evergage classes, the idea is to encourage clients to use evergage class names for organizing their code and maintaining consistency across their templates. My aim for Global templates, however, is to have the same class style definitions for all Evergage classes for consistency. 

I am finalizing my [Evergage Templates Style Guide and Coding Conventions](https://docs.google.com/document/d/1tkisQNYXttDkaWl3pkdgNTsL-mmTWj9xx3QNQs4VYH4/edit) and will be including all of this information in there as well. 

### Things to check for:
* **Visuals:** Does it look good across different client sites? To test this, you can use this snippet https://gist.github.com/nriserEG/b5d83191f7ecad8fda832e99bc1da150 in the Chrome inspector tool that injects the minified template HTML and CSS code (the template will appear at the very top of the page you test on).
* **Deferred rendering:** Is the rendering of the template deferred until the content zone element is loaded on page? 
* **Flicker defense:** Is the template flicker-defensed? Promise must be returned in the apply function for flicker defense to work. 
* **Campaign stats tracking:** Do impression stats and click stats (upon clicking the CTA) fire for both the control and test experience?
* **Template-specific documentation**: Is there sufficient information for a template developer to use the global template and customize for their own use?
* **WCAG compliant:** Does the template contain ARIA attributes and use semantic elements for web accessibility? 
** FYI for this template specifically, I included aria-label attributes and adds fallback background colors in case the background image fails to load.
** You can use the WAVE Chrome extension tool https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh for a quick check. 
* **Mobile/smaller screen responsiveness:** Does the campaign generally look good on smaller screens?

**Important note:** In order to test **Flicker Defense** properly, we want to test the template on a site that already loads Beacon v16. The nto demo site (northerntrailoutfitters.com) does not have the beacon loaded yet and it’s not clear when it’ll be added. Per @ghinkle, we may want to add Beacon v16 on the Dev Docs site to test the template for Flicker Defense. Not sure what the status is on this (@kcsizmadiaEG / @avkieu will give an update on QA approach for this).

#### Template: 
- You can view the demo template in eng1:globaltemplates, **template name: [WIP] Global - Hero Banner with Call-to-Action**. 

#### Campaign: 
- You can view the demo campaign using this template in eng1:globaltemplates, **campaign name: [WIP] Global - Hero Banner with Call-to-Action**
- Sample nto test link: https://www.northerntrailoutfitters.com/default/homepage?evergageTestMessages=7GTJn

Note: If you test on nto (northerntrailoutfitters.com), you will have to inject the beacon.

## Design decisions

- I used the northerntrailoutfitters.com site to develop the template, so the content (background image & text) is meant to match the nto use case. 
- Background image: I chose [this background image](https://www.northerntrailoutfitters.com/dw/image/v2/BDPX_PRD/on/demandware.static/-/Library-Sites-NTO-SFRASharedLibrary/default/dw37eb7010/images/homepage/home-banner-hero-04-1905-600.jpg) to use in this global template because it's visually the simplest and the least distracting out of all nto images that I could find, and it works well with both light text and dark text in terms of **WCAG** compliant color contrast ratio.
- Fallback background color: I added fallback background colors to the background image in case the image does not load so that the header texts are still accessible. Both the text color and the background color will change based on the Color Theme option chosen.
- Button color: I wanted to stick to the same button color as the color used for buttons on the nto site, but I noticed that the color contrast between the foreground and background colors were not **WCAG** compliant (contrast is too low -- works only if font is bolded and size is 18px or larger) so I intentionally used a darker shade of blue (same color used on the Salesforce.com site), which is **WCAG** compliant. (See https://aremycolorsaccessible.com/)

### CODE REVIEW

**Comment by nriserEG** [comment]
Few new updates:
- Removed the clickthrough attribute from anchor tag to match specs that Dan, Alex, and I agreed upon. There is a ticket for the updates here, including documentation: https://help.evergage.com/browse/EVG-11659
- Per Greg, I added a png image file of a screen shot of the template. 
- Updated the Hero Banner template section on README

Also added a section on "Design decisions" in the PR description to add some insight as to decisions I made.

File: hero-banner-cta/template.hbs [hunk 1]
```diff
@@ -1,12 +1,41 @@
...
+data-evg-experience-id="{{experience}}" data-evg-user-group="{{userGroup}}" aria-label="Hero banner" 
+style="background-image:url({{imageURL}});">
+    <div class="evg-container">
+        <h1 class="evg-header evg-h1 evg-display-2 evg-text">
+            {{header}}
+        </h1>
+        <h2 class="evg-subheader evg-h2 evg-text">
+            {{subheader}}
+        </h2>
+        <a class="evg-cta evg-btn evg-btn-primary evg-btn-lg" href="{{ctaUrl}}" aria-label="Shop {{header}}">
```
Comment (aspatti1257) [inline hero-banner-cta/template.hbs:38]: Doesn't it need:
`data-evg-clickthrough`
In order to track clickthroughs properly?

Comment (nriserEG) [reply hero-banner-cta/template.hbs:38]: @aspatti1257 
The clickthrough attribute was removed from the anchor tag to match new specs that Dan, Alex, and I agreed upon. Updates are currently in Review: https://help.evergage.com/browse/EVG-11659 / https://github.com/evergage/evergage-product/pull/1123

**Comment by nriserEG** [comment]
@aspatti1257 thanks for your review.

The clickthrough attribute was removed from the anchor tag to match new specs that Dan, Alex, and I agreed upon:

> Any Evergage campaign (has data-evg-experience-id attribute) will automatically track child <a> tags
> An <a> tag can negate automatic clickthrough tracking with the data-evg-ignore-clickthrough attribute
> All other elements can track clickthroughs if they have the data-evg-clickthrough attribute

Andy is working on making the update: 
- JIRA: https://help.evergage.com/browse/EVG-11659
- PR: evergage/evergage-product#1194

File: hero-banner-cta/template.ts [hunk 2]
```diff
@@ -1,28 +1,35 @@
...
 
-    style: "Dark on Light" | "Light on Dark";
+    @hidden(true)
+    contentZone: string = "Homepage | Hero";
+  
+    @options([
+        {label: "Dark text on light background", className: "evg-dark-on-light"},
+        {label: "Light text on dark background", className: "evg-light-on-dark"}
+    ])
+    colorTheme: ColorTheme;
```
[the remark this answers is not in the document]
Comment (nriserEG) [reply hero-banner-cta/template.ts]: @aspatti1257 Default added!

**Pull request merged**